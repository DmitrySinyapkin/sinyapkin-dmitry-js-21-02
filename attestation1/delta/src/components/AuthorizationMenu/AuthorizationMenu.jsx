import React from "react";
import './AuthorizationMenu.css';
import { Link } from "react-router-dom";
import { connect } from 'react-redux';
import { bindActionCreators } from "redux";
import * as actions from '../../actions/authorization';
import { Avatar } from 'antd';
import { useTranslation } from "react-i18next";

const AuthorizationMenu = ({ authUser, authorized, cancelUser }) => {
    const { t } = useTranslation();

    return ( authorized ? 
                            <div className="auth-menu">
                                <Link to={`/users/${authUser.id}`}>
                                    <div className="auth-menu__first user">
                                        <Avatar shape="circle" src={authUser.picture} />
                                        <div className="name">{authUser.firstName}</div>
                                    
                                    </div>
                                </Link>
                                <div className="auth-menu__second"><a href="" onClick={cancelUser}>{t('login.cancel')}</a></div>
                            </div>
                        :
                            <div className="auth-menu">
                                <div className="auth-menu__first"><Link to="/auth">{t('login.enter')}</Link></div>
                                <div className="auth-menu__second"><Link to="/reg">{t('login.reg')}</Link></div>
                            </div>
    );
}

export default connect(
    (state) => ({
        authUser: state.auth.authUser,
        authorized: state.auth.authorized,
    }),
    (dispatch) => bindActionCreators(actions, dispatch),
)(AuthorizationMenu);

