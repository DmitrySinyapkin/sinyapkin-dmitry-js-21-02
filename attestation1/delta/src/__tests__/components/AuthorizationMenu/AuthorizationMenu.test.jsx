import React from "react";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AuthorizationMenu from "../../../components/AuthorizationMenu/AuthorizationMenu";
import { MemoryRouter } from "react-router";
import { render } from 'enzyme';

const mockStore = configureStore([thunk]);

describe('authorization menu', () => {
    test('snapshot', () => {
        const store = mockStore({
            auth: {
                authUser: {
                    picture: 'url',
                    firstName: 'Vasya'
                },
                authorized: false,
            }
        })
        const wrapper = render(<MemoryRouter><AuthorizationMenu store={store} /></MemoryRouter>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render authorized user menu', () => {
        const store = mockStore({
            auth: {
                authUser: {
                    picture: 'url',
                    firstName: 'Vasya'
                },
                authorized: true,
            }
        })
        const wrapper = render(<MemoryRouter><AuthorizationMenu store={store} /></MemoryRouter>);
        expect(wrapper.find('div.name').text()).toBe('Vasya');
    });
});
