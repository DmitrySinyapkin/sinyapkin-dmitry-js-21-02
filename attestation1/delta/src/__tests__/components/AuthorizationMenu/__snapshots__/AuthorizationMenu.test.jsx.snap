// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorization menu should render menu 1`] = `
<div
  class="auth-menu"
>
  <div
    class="auth-menu__first"
  >
    <a
      href="/auth"
    >
      login.enter
    </a>
  </div>
  <div
    class="auth-menu__second"
  >
    <a
      href="/reg"
    >
      login.reg
    </a>
  </div>
</div>
`;

exports[`authorization menu snapshot 1`] = `
<div
  class="auth-menu"
>
  <div
    class="auth-menu__first"
  >
    <a
      href="/auth"
    >
      login.enter
    </a>
  </div>
  <div
    class="auth-menu__second"
  >
    <a
      href="/reg"
    >
      login.reg
    </a>
  </div>
</div>
`;
